### ISSUE TITLE
Sentry database corruption? Authentication and exporting breaking

### ISSUE THREAD

**Issue by failattu** [getsentry/self-hosted#1259]
### Version

21.6.3

### Steps to Reproduce

1. We have 200 gigabytes of data that had postgres val missmatch for somereason so we had to run https://www.postgresql.org/docs/9.6/app-pgresetxlog.html 
2. Having our previous version 21.5.1 booted SAML authentication does not work for 2/3 people we have tested so far
3. Bunch of unique constraint complains? maybe some migration script not running

### Expected Result

authentication works 

### Actual Result

Authentication and export fail on error

Creating sentry_onpremise_web_run ... done
>> Beginning export
>> Skipping model <Broadcast>
>> Skipping model <CommitAuthor>
>> Skipping model <FileBlob>
>> Skipping model <File>
>> Skipping model <FileBlobIndex>
>> Skipping model <FileBlobOwner>
>> Skipping model <ProjectDebugFile>
>> Skipping model <DeletedOrganization>
>> Skipping model <DeletedProject>
>> Skipping model <DeletedTeam>
>> Skipping model <EventAttachment>
>> Skipping model <EventUser>
>> Skipping model <ExternalIssue>
>> Skipping model <GroupCommitResolution>
>> Skipping model <GroupLink>
>> Skipping model <GroupRedirect>
>> Skipping model <GroupRelease>
>> Skipping model <IdentityProvider>
>> Skipping model <IntegrationExternalProject>
>> Skipping model <LatestRepoReleaseEnvironment>
>> Skipping model <NotificationSetting>
>> Skipping model <ProjectPlatform>
>> Skipping model <PullRequest>
>> Skipping model <ReleaseFile>
>> Skipping model <ScheduledDeletion>
>> Skipping model <ScheduledJob>
>> Skipping model <UserAvatar>
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/db/models/fields/related_descriptors.py", line 178, in __get__
    rel_obj = getattr(instance, self.cache_name)
AttributeError: 'UserEmail' object has no attribute '_user_cache'
Export fails: 
    self.model._meta.object_name
sentry.models.user.DoesNotExist: User matching query does not exist.
ERROR: 1

SAML:
    auth_identity.update(user=user, data=identity.get("data", {}))
AttributeError: 'NoneType' object has no attribute 'update'
django.db.utils.IntegrityError: IntegrityError('duplicate key value violates unique constraint "sentry_authidentity_auth_provider_id_ident_c18fcc44_uniq"\nDETAIL:  Key (auth_provider_id, ident)=(1, kalle.sirkesalo@eficode.com) already exists.\n',)
SQL: INSERT INTO "sentry_authidentity" ("user_id", "auth_provider_id", "ident", "data", "last_verified", "last_synced", "date_added") VALUES (%s, %s, %s, %s, %s, %s, %s) RETURNING "sentry_authidentity"."id"
22:02:14 [ERROR] django.request: Internal Server Error: /saml/acs/sentry/ (status_code=500 request=<WSGIRequest: POST '/saml/acs/sentry/'>)

Select * from sentry_authidentity; returns 0 lines

Authentication is now broken

**Comment by chadwhitacre** [comment]
Hrm, seems like you're deep in it @failattu. Your steps to reproduce are precise enough to follow along with, though. Can you update them to be more specific?

**Comment by failattu** [comment]
@chadwhitacre I wish we could explain HOW we ended up here, we know our upgrade corrupted the DB -> leading to need to fix it.
Now we have managed to recover one of the PSQL's in "proper state today" and managed to login to it. This lead to statistics blowing up:
"but there is no activities or statistics (those were probably kept in kafka, which was not migrated for now, we’ve just wanted to boot up anything)"
Getting us:
2022-01-19T12:57:44Z [relay_server::actors::upstream] WARN: Network outage, scheduling another check in 60s
2022-01-19T12:58:31Z [relay_server::actors::upstream] ERROR: authentication encountered error: upstream request returned error 503 Service Unavailable
  caused by: no error details
2022-01-19T12:58:44Z [relay_server::actors::upstream] WARN: Network outage, scheduling another check in 60s
2022-01-19T12:59:31Z [relay_server::actors::upstream] ERROR: authentication encountered error: upstream request returned error 503 Service Unavailable
  caused by: no error details
2022-01-19T12:59:44Z [relay_server::actors::upstream] WARN: Network outage, scheduling another check in 60s
2022-01-19T12:59:52Z [relay_server::endpoints::common] ERROR: error handling request: failed to queue envelope
  caused by: Too many envelopes (event_buffer_size reached)
  
  I.e. it can't relay, so it buffers a bit, then responds with the "too many envelopes" when it's full


where is the relay supposed to connect to. Redis? Kafka?
relay:
  upstream: "http://web:9000/"
  host: 0.0.0.0
  port: 3000
logging:
  level: WARN
processing:
  enabled: true
  kafka_config:
    - {name: "bootstrap.servers", value: "kafka:9092"}
    - {name: "message.max.bytes", value: 50000000} # 50MB
  redis: redis://redis:6379
  geoip_path: "/geoip/GeoLite2-City.mmdb"
  
  with tcpdup we catched that our http_proxy was being fed to containers leading to internal traffic in containers flowing to said proxy, we fixed it with:
 getting rid of those vars by a removing them from everywhere (bash session, systemd, .docker/config.json)

Transactions are getting accepted in Sentry again. 
So we got it working, but it was a journey and would not recommend it, still don't know what corrupted in the database to prevent the logins, but thank god we had a "backup" available.
